libevmcore/Assembly.cpp

```diff
@@ -156,6 +166,134 @@ ostream& Assembly::stream(ostream& _out, string const& _prefix, StringMap const&
 	return _out;
 }
 
+Json::Value Assembly::createJsonValue(string _name, int _locationX, int _locationY, string _value, string _jumpType) const
+{
+	Json::Value value;
+	assert(!_name.empty());
+	value["name"] = _name;
+	value["locationX"] = _locationX;
+	value["locationY"] = _locationY;
+	if (!_value.empty())
+		value["value"] = _value;
+	if (!_jumpType.empty())
+		value["jumpType"] = _jumpType;
+	return move(value);
```

Comment:
As far as I know, using move when returning prevents the compiling from doing some optimizations.